src/gui/editors/AutomationEditor.cpp

```diff
@@ -1587,12 +1587,36 @@ void AutomationEditor::drawLevelTick(QPainter & p, int tick, float value)
 
 		p.fillRect( x, y_start, rect_width, rect_height, currentColor );
 	}
-
+#ifdef LMMS_DEBUG
 	else
 	{
 		printf("not in range\n");
 	}
+#endif
+}
+
+
+
 
+// center the vertical scroll position on the first object's value
+void AutomationEditor::centerTopBottomScroll()
+{
+	// default to the m_scrollLevel position
+	int pos = (int) m_scrollLevel;
+	// If a pattern exists...
+	if (m_pattern) {
+		// get time map of current pattern
+		timeMap & time_map = m_pattern->getTimeMap();
+		// If time_map is not empty...
+		if ( !time_map.empty() ) {
+			// get the first object
+			timeMap::iterator it = time_map.begin();
+			// set the position to the inverted value ((max + min) - value)
+			// If we set just (max - value), we're off by m_pattern's minimum
+			pos = m_pattern->getMax() + m_pattern->getMin() - (int) it.value();
```

Comment:
Couldn't you use timeMap.begin() directly instead of the iterator?

Reply:
Also we should avoid spaces in braces for new code. Have a look at our coding convention.

Reply:
> Couldn't you use timeMap.begin() directly instead of the iterator?

Ah, you're right. I was copying other code for this. I'll fix it.

> Also we should avoid spaces in braces for new code. Have a look at our coding convention.

I've been copying the style currently there thinking it's what's preferred, and I see spaces in the code all over. I assume this is a new convention going forward? Is there a need to go through and update style?

Reply:
In the past we used spaces. It's common sense that we change it when we touch the code. But don't touch code if it's not necessary.